src/fst.rs

```diff
@@ -318,80 +318,86 @@ pub(crate) fn push_zeros(vec: &mut Vec<u8>, len: usize) {
     }
 }
 
-fn convert_scope_tpe(tpe: FstScopeType) -> ScopeType {
-    match tpe {
-        FstScopeType::Module => ScopeType::Module,
-        FstScopeType::Task => ScopeType::Task,
-        FstScopeType::Function => ScopeType::Function,
-        FstScopeType::Begin => ScopeType::Begin,
-        FstScopeType::Fork => ScopeType::Fork,
-        FstScopeType::Generate => ScopeType::Generate,
-        FstScopeType::Struct => ScopeType::Struct,
-        FstScopeType::Union => ScopeType::Union,
-        FstScopeType::Class => ScopeType::Class,
-        FstScopeType::Interface => ScopeType::Interface,
-        FstScopeType::Package => ScopeType::Package,
-        FstScopeType::Program => ScopeType::Program,
-        FstScopeType::VhdlArchitecture => ScopeType::VhdlArchitecture,
-        FstScopeType::VhdlProcedure => ScopeType::VhdlProcedure,
-        FstScopeType::VhdlFunction => ScopeType::VhdlFunction,
-        FstScopeType::VhdlRecord => ScopeType::VhdlRecord,
-        FstScopeType::VhdlProcess => ScopeType::VhdlProcess,
-        FstScopeType::VhdlBlock => ScopeType::VhdlBlock,
-        FstScopeType::VhdlForGenerate => ScopeType::VhdlForGenerate,
-        FstScopeType::VhdlIfGenerate => ScopeType::VhdlIfGenerate,
-        FstScopeType::VhdlGenerate => ScopeType::VhdlGenerate,
-        FstScopeType::VhdlPackage => ScopeType::VhdlPackage,
-        FstScopeType::AttributeBegin
-        | FstScopeType::AttributeEnd
-        | FstScopeType::VcdScope
-        | FstScopeType::VcdUpScope => unreachable!("unexpected scope type!"),
+impl From<FstScopeType> for ScopeType {
```

Comment:
Implementing this trait for `ScopeType` (which is `pub`) makes this functionality part of the public API. It also pulls the `FstScopeType` into the public API. My major goal when building libraries is to keep the public API as small as possible. In general I only accept API expansions if there is a use case, e.g., if surfer needs to do something that cannot be done with the current API.